File: go/06-io-operations/read_input.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	fmt.Print("Enter your name: ")
	scanner.Scan()
	name := scanner.Text()

	fmt.Print("Enter your age: ")
	scanner.Scan()
	age := scanner.Text()

	fmt.Printf("Hello, %s! You are %s years old.\n", name, age)
}

File: haskell/06-io-operations/read_input.hs
main :: IO ()
main = do
    putStrLn "What is your name?"
    name <- getLine                 -- bind the String out of IO String

    putStrLn "How old are you?"
    ageStr <- getLine
    let age = read ageStr :: Int    -- pure parsing with a type annotation

    putStrLn ("Hello, " ++ name ++ "!")
    putStrLn ("Next year you will be " ++ show (age + 1))
